### DISCUSSION TITLE
Themes

### DISCUSSION THREAD

**Discussion by blondie63** [Laravel-Backpack/community-forum#648] [Q&A (Help)] [answered]
Thinking from a multi-theme perspective, as the backpack demo does, I have a few questions:
Documentation on these two functions ?
backpack_theme_config()
backpack_view()

To define menus in BKP5 used code like this:
```<li class="nav-title">{{ trans('backpack::base.user_level') }}</li>
    <li class='nav-item'>
        <a id="supervisor-link" class='nav-link' href="/admin/dashboard">
            <i class='nav-icon la la-eye'></i>{{ trans('backpack::base.supervisor') }}
        </a>
    </li>
```

As I see the demo uses this:
```
<x-backpack::menu-dropdown title="Add-ons" icon="la la-puzzle-piece">
    <x-backpack::menu-dropdown-header title="News" />
    <x-backpack::menu-dropdown-item title="Articles" icon="la la-newspaper-o" :link="backpack_url('article')" />
    <x-backpack::menu-dropdown-item title="Categories" icon="la la-list" :link="backpack_url('category')" />
    <x-backpack::menu-dropdown-item title="Tags" icon="la la-tag" :link="backpack_url('tag')" />
```

Is it better to use the second one?

Thank you

**Comment by pxpm** [comment, marked as answer]
Indeed @blondie63 using the blade components allow each theme to output the needed html to work the way intended by the theme developer. 

This are the docs on available components: https://backpackforlaravel.com/docs/6.x/base-components#available-components-1

Each theme may or may not overwrite those components like in tabler for example: https://github.com/Laravel-Backpack/theme-tabler/blob/main/resources/views/components/menu-dropdown.blade.php

Take into consideration that we are aware of some limitations at the moment with the components, one of them the nested menus that we are working out here: https://github.com/Laravel-Backpack/CRUD/pull/5289

About the `backpack_theme_config()` is documented here: https://backpackforlaravel.com/docs/6.x/base-themes#theme-configs-1 and the `backpack_view()` here: https://backpackforlaravel.com/docs/6.x/base-themes#theme-view-fallbacks-1

Cheers
